books/spiders/book.py

```diff
@@ -0,0 +1,58 @@
+from collections.abc import Generator
+from typing import Optional, Union
+
+import scrapy
+
+from scrapy.http import Response
+
+
+class BookSpider(scrapy.Spider):
+    name = "book"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
```

Comment:
Pls check the officials docs and follow the recommendations :

https://docs.scrapy.org/en/latest/intro/tutorial.html